plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -147,8 +150,17 @@ public ConnectorSplitSource getSplits(
                         log.debug("EXPECTED_UPPER_REGION_KEY[%d] : %s", i, Bytes.toStringBinary(scans.get(i).getAttribute(EXPECTED_UPPER_REGION_KEY)));
                     }
                 }
-                inputSplits.add(new PhoenixInputSplit(scans, regionSize, regionLocation));
+                /*
+                 * Handle parallel execution explicitly in Trino rather than internally in Phoenix.
+                 * Each split is handled by a single ConcatResultIterator
+                 * (See PhoenixClient.getResultSet(...))
+                 */
+                for (List<Scan> splitScans : Lists.partition(scans, maxScansPerSplit)) {
+                    inputSplits.add(new PhoenixInputSplit(splitScans, regionSize, regionLocation));
+                }
             }
+            // shuffle to distribute load across HBase regions
```

Comment:
The order of splits in list shouldn't make a difference. See `io.trino.execution.scheduler.SplitPlacementPolicy#computeAssignments`. Splits are `Set` there

Reply:
Thanks. So the shuffle is unneeded, I'll remove it.